exchanges/subscription/template_test.go

```diff
@@ -84,13 +84,18 @@ func TestExpandTemplates(t *testing.T) {
 	}
 	equalLists(t, exp, got)
 
+	// Users can specify pairs which aren't available, even across diverse assets
+	// Use-case: Coinbasepro user sub for futures BTC-USD would return all BTC pairs and all USD pairs, even though BTC-USD might not be enabled or available
```

Comment:
I'm fine with the changes regarding pair validation, but I am a bit confused with this.
This PR does not mention #1480 as otherwise we don't have CBP futures support. The other thing is that GCT will do a purge of any unlisted pairs:
Before:
```json
  "futures": {
      "assetEnabled": true,
      "enabled": "BTC-USD",
      "available": "..."
     },
```
After:
`Exchange CoinbasePro: [futures] Removing enabled pair(s) [BTC-USD] from enabled pairs list, as it isn't located in the available pairs list.`

```json
     "futures": {
      "assetEnabled": true,
      "enabled": "BIT-25OCT24-CDE",
      "available": "..."
     },
```

`[INFO]  | EXCHANGE | 23/10/2024 12:30:29 | CoinbasePro Asset type FUTURES:
	 Enabled pairs: [BIT-25OCT24-CDE]`

Because of this, I want to know what scenario you are speaking of and how to check it so that I can be happy the situation is resolved 🎉

Reply:
I *think* #1480 would be the right reference.
@cranktakular  raised it with me directly.
> This came up in a Coinbase test for a websocket channel which doesn't distinguish between asset types; a request for "BTC-USD" will return all spot, perpetual futures, and expiring futures items associated with those two currencies.

So I don't know how he's going to make the pair end up in the subscription pairs.
He's free to have it happen during the template called from GenerateSubscriptions, or set the subscription that way programmatically using GCT as a library.

This PR just aims to get out of the user's way and not introduce unnecessary checks.

Reply:
Thanks for providing details! I'll seek clarification in 1480